Add "Metadata saved" notification

Arnei opened the pull request:
Adds a short notification whenever the user clicks on "Save" in the metadata tab.

![Bildschirmfoto vom 2025-05-20 11-52-55](https://github.com/user-attachments/assets/88e78f8d-8ade-4a43-8883-256725ee938d)

### How to test this

Can be tested as usual. Try saving some metadata!

Arnei committed:
Add "Metadata saved" notification

Adds a short notification whenever the user
clicks on "Save"
in the metadata tab.

JulianKniephoff commented on src/components/events/partials/ModalTabsAndPages/DetailsMetadataTab.tsx:
```diff
@@ -52,6 +54,13 @@ const DetailsMetadataTab = ({
+			duration: 3,
+			parameter: undefined,
+			context: NOTIFICATION_CONTEXT,
+		}));
```
Should this not wait for the above action to complete? Also what if it fails? 🤔

Arnei replied:
How dare you not letting me get away with my half-assed implementation >.> Should properly respect action states now.

Arnei committed:
Improve "Metadata saved" notification

Only display the notification after the related
web request has actually successfully completed.
Also display a warning in case the related web request fails.

JulianKniephoff commented on src/components/events/partials/ModalTabsAndPages/DetailsMetadataTab.tsx:
```diff
@@ -52,6 +54,13 @@ const DetailsMetadataTab = ({
+			parameter: undefined,
+			context: NOTIFICATION_CONTEXT,
+		}));
 	};
```
Wait, shouldn't something like this suffice? 👀

```suggestion
	const handleSubmit = async (values: { [key: string]: any }, catalog: MetadataCatalog) => {
		try {
			await dispatch(updateResource({id: resourceId, values, catalog}));
			dispatch(addNotification({
				type: "info",
				key: "METADATA_SAVED",
				duration: 3,
				parameter: undefined,
				context: NOTIFICATION_CONTEXT,
			}));
		} catch (_) {
			dispatch(addNotification({
				type: "warning",
				key: "METADATA_NOT_SAVED",
				duration: 3,
				parameter: undefined,
				context: NOTIFICATION_CONTEXT,
			}));
		}
	};
```

Arnei replied:
...fair enough. My intuition told me to not do this, but now that I'm thinking about it more I'm struggling to come up with any good reasons on why this would be a bad approach.

Arnei committed:
Revert "Improve "Metadata saved" notification"

This reverts commit 5394f1b910f9b7d0f727fb28223ada907629b704.

Arnei committed:
Improve "Metadata saved" notification

Only display the notification after the related
web request has actually successfully completed.
Also display a warning in case the related web request fails.

JulianKniephoff commented on src/components/events/partials/ModalTabsAndPages/DetailsMetadataTab.tsx:
```diff
@@ -50,8 +52,27 @@ const DetailsMetadataTab = ({
+					parameter: undefined,
+					context: NOTIFICATION_CONTEXT,
+				}));
+			})
```
Interesting! I would prefer

```suggestion
	const handleSubmit = async (values: { [key: string]: any }, catalog: MetadataCatalog) => {
		try {
			await dispatch(updateResource({ id: resourceId, values, catalog })).unwrap();
			
			dispatch(addNotification({
				type: "info",
				key: "METADATA_SAVED",
				duration: 3,
				parameter: undefined,
				context: NOTIFICATION_CONTEXT,
			}));
		} catch {
			dispatch(addNotification({
				type: "warning",
				key: "METADATA_NOT_SAVED",
				duration: 3,
				parameter: undefined,
				context: NOTIFICATION_CONTEXT,
			}));
		}
	};
```

But I'd we willing to let that slide as a matter of taste for teaching me about the necessity of `unwrap` here. :D

**However**, if you don't `await`, you don't need `async`, either.

JulianKniephoff commented on src/components/events/partials/ModalTabsAndPages/DetailsMetadataTab.tsx:
```diff
@@ -50,8 +52,27 @@ const DetailsMetadataTab = ({
+					type: "info",
+					key: "METADATA_SAVED",
+					duration: 3,
+					parameter: undefined,
```
While we're at it: `key: undefined` is always* redundant.

JulianKniephoff commented on src/components/events/partials/ModalTabsAndPages/DetailsMetadataTab.tsx:
```diff
@@ -50,8 +52,27 @@ const DetailsMetadataTab = ({
+					type: "warning",
+					key: "METADATA_NOT_SAVED",
+					duration: 3,
+					parameter: undefined,
```
See above.

Arnei committed:
Remove some superfluous code

None of these are necessary

Arnei committed:
Merge branch 'develop' into metadata-saved-notification

Arnei committed:
Merge branch 'develop' into metadata-saved-notification

JulianKniephoff reviewed (approved):
LGTM now 😊
